src/commands/sourcemaps/upload.ts

```diff
@@ -141,47 +137,34 @@ export class UploadCommand extends Command {
   }
 
   // Fills the 'repository' field of each payload with data gathered using git.
-  private addRepositoryDataToPayloads = async (payloads: Payload[]) => {
+  private addRepositoryDataToPayloads = async (payloads: Sourcemap[]) => {
     const repositoryData = await getRepositoryData(await newSimpleGit(), this.context.stdout, this.repositoryURL)
     if (repositoryData === undefined) {
       return
     }
     await Promise.all(
       payloads.map(async (payload) => {
-        payload.gitRepositoryPayload = this.getRepositoryPayload(repositoryData, payload.sourcemapPath)
-        payload.gitRepositoryURL = repositoryData.remote
-        payload.gitCommitSha = repositoryData.hash
+        const repositoryPayload = this.getRepositoryPayload(repositoryData!, payload.sourcemapPath)
+        payload.addRepositoryData({
+          gitCommitSha: repositoryData.hash,
+          gitRepositoryPayload: repositoryPayload,
+          gitRepositoryURL: repositoryData.remote,
+        })
```

Comment:
I introduced a new interface GitData to group all git-related information into a single object (the object is either present entirely or not, no real use case to have these fields partially defined).